src/test/java/org/example/bookshop/controller/BookControllerTest.java

```diff
@@ -0,0 +1,259 @@
+        bookDto.setCategoryIds(Set.of(FIRST_ID));
+
+        requestDto.setTitle(BOOK_TITLE);
+        requestDto.setAuthor(BOOK_AUTHOR);
+        requestDto.setIsbn(BOOK_ISBN);
+        requestDto.setPrice(BOOK_PRICE);
+        requestDto.setCategories(categories);
+
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            List<String> scriptPaths = List.of(
+                    "database/categories/add-category-to-categories-table.sql",
+                    "database/books/add-book-to-books-table.sql",
+                    "database/categories/add-category-for-books_categories-table.sql"
+            );
+            for (String scriptPath : scriptPaths) {
+                ScriptUtils.executeSqlScript(connection, new ClassPathResource(scriptPath));
+            }
+        }
+    }
+
+    @AfterEach
+    public void afterEach(@Autowired DataSource dataSource) throws SQLException {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            List<String> scriptPaths = List.of(
+                    "database/categories/delete-all-categories-from-categories-table.sql",
+                    "database/books/delete-all-books-from-books-table.sql"
+            );
+            for (String scriptPath : scriptPaths) {
+                ScriptUtils.executeSqlScript(connection, new ClassPathResource(scriptPath));
+            }
+        }
+    }
+
+    @Test
+    @WithMockUser(roles = {"ADMIN"})
+    @DisplayName("Create a new book")
+    void createBook_ValidRequestDto_ReturnCreateBookRequestDto() throws Exception {
+        // Given
+        CreateBookRequestDto saveRequestDto = new CreateBookRequestDto();
+        saveRequestDto.setTitle(BOOK_TITLE + UNIQUE_PARAM);
+        saveRequestDto.setAuthor(BOOK_AUTHOR + UNIQUE_PARAM);
+        saveRequestDto.setIsbn(BOOK_ISBN + UNIQUE_PARAM);
+        saveRequestDto.setPrice(BOOK_PRICE);
+        saveRequestDto.setCategories(categories);
+        String jsonRequest = objectMapper.writeValueAsString(saveRequestDto);
+        // When
+        MvcResult result = mockMvc.perform(
+                post("/books")
+                .content(jsonRequest)
+                .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isCreated())
+                .andReturn();
+        // Then
+        BookDto actual = objectMapper.readValue(
+                result.getResponse().getContentAsString(), BookDto.class);
+        EqualsBuilder.reflectionEquals(saveRequestDto, actual);
+    }
+
+    @Test
+    @Sql(scripts = {"classpath:database/books/add-two-books-to-books-table.sql",
+            "classpath:database/categories/add-second-category-to-categories-table.sql",
+            "classpath:database/categories/add-two-category-to-books_categories-table.sql"},
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
```

Comment:
Let's move all @Sql scripts above class, it will help to understand test methods much easier, because we have the same data and state for each method. Keep it simple :)